samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -551,8 +553,23 @@ private static void runClusterBasedJobCoordinator(String[] args) {
         throw new SamzaException(e);
       }
 
-      ClusterBasedJobCoordinator jc = createFromConfigLoader(submissionConfig);
-      jc.run();
+      ApplicationConfig appConfig = new ApplicationConfig(submissionConfig);
+
+      if (appConfig.getAppMainClass().isPresent()) {
```

Comment:
Please add some comments about the use cases here for Beam. what's the expected main/args being passed in and what are we going to do with them.

Reply:
Added